docs/examples/junit4/generic/src/test/java/org/testcontainers/utility/TestSpecificImageNameSubstitutor.java

```diff
@@ -0,0 +1,25 @@
+package org.testcontainers.utility;
+
+/**
+ * An {@link ImageNameSubstitutor} which makes it possible to use fake image names in
+ * {@link generic.ImageNameSubstitutionTest}. This implementation simply reverses a fake image name when presented, and
+ * is hardcoded to act upon the specific fake name in that test.
+ */
+public class TestSpecificImageNameSubstitutor extends ImageNameSubstitutor {
+
+    private final DefaultImageNameSubstitutor defaultImageNameSubstitutor = new DefaultImageNameSubstitutor();
+
+    @Override
+    public DockerImageName apply(final DockerImageName original) {
+        if (original.equals(DockerImageName.parse("registry.mycompany.com/mirror/mysql:8.0.22"))) {
+            return defaultImageNameSubstitutor.apply(DockerImageName.parse("mysql"));
+        } else {
+            return defaultImageNameSubstitutor.apply(original);
```

Comment:
is it still needed, with the latest change where we always apply the default?

It is a nit but, since we refer to it from the docs, serves as an important example :)

Reply:
I just realized... `DefaultImageNameSubstitutor` is internal, so we definitely should avoid using it in this example :) I would even move it to some other package to ensure that everyone can write a custom substitutor

Reply:
Argh, good point. Removed, which also means that I can move this class back out of the `org.testcontainers.utility` package because there's no longer a dependency on the default substitutor.